pulsar-functions/proto/src/main/proto/Function.proto

```diff
@@ -98,6 +98,7 @@ message SourceSpec {
      * already present in the server */
     string builtin = 8;
     string subscriptionName = 9;
+    bool cleanupSubscription = 11;
```

Comment:
while you are trying to set cleanupSubscription in FunctionConfig.java to true if not specified, we need to also do similar things when we get the Function/Source/Sink Config in the worker side of things(since things could be submitted outside of cmdline tool). 
I think a better way is to have a dontcleanSubscription everywhere. The fact that the default value of boolean is false will immensly help in code complexity that way.

Reply:
@srkukarni I am setting the Sink and Function config on the worker side to default to true via the SinkConfigUtils and FunctionConfigUtils.  Also this property is not relevant to Sources

Reply:
@srkukarni I purposely designed the flag this way for backwards compatibility purposes.  A function that is running with an older version of the code/metadata will still not cleanup subscriptions as that was the behavior before